src/AlgebraicGeometry/Schemes/PrincipalOpenSubset/Objects/Types.jl

```diff
@@ -33,8 +33,21 @@
   function PrincipalOpenSubset(X::AbsAffineScheme, R::Ring, f::Vector{T};
       check::Bool=true
     ) where {T<:RingElem}
-    d = prod(length(f) > 0 ? f : one(OO(X)))
-    return PrincipalOpenSubset(X, R, lifted_numerator.(f), d, check=check)
+    @check begin
+        d = prod(length(f) > 0 ? f : one(OO(X)))
+        U = spec(R)
+        U == hypersurface_complement(X, d)
+    end
+    return new{base_ring_type(X), ring_type(U), typeof(X)}(X, U, lifted_numerator.(f))
```

Comment:
this looks fishy to me. `U` is not defined if check=false
Also why the change? The code looks unnecessarily complicated to me. And how is it guaranteed that the output is a `PrincipalOpenSubset` ?

Reply:
You're right. I'll look into this.